src/group_theory/quotient_group.lean

```diff
@@ -412,4 +413,42 @@ monoid_hom.to_mul_equiv
 
 end third_iso_thm
 
+
+section trivial
+
+/-These first few lemmas should maybe go somewhere else, where?-/
+
+lemma quot_by_top_is_subsingleton : subsingleton (quotient_group.quotient (⊤ : subgroup G)) :=
+trunc.subsingleton
+
+lemma left_rel_triv (H : subgroup G) (h : quotient_group.left_rel H = ⊤ ) :
+  ∀ x y : G , x⁻¹ * y ∈ H :=
+begin
+  let s := (quotient_group.left_rel H).r,
+  have h2 : ∀ x y : G, s x y ↔ x⁻¹ * y ∈ H, by {intros x y, refl},
+  intros x y,
+  have h3 := h2 x y,
+  rw ← h3,
+  simp_rw s,
+  rw h,
+  tauto,
+end
+
+/-- If the quotient by a subgroup gives a singleton then the subgroup is the whole group. -/
+lemma subgroup_eq_top_of_subsingleton (H : subgroup G) (h : subsingleton
+  (quotient_group.quotient H)) :
+  H = ⊤ :=
+begin
+  ext1,
+  simp only [subgroup.mem_top, iff_true] at *,
+  rw quotient_group.quotient at h,
+  have h2 := (setoid.subsingleton_quotient_iff_eq_top).1 h,
+  have h3 := left_rel_triv H h2,
+  have h4 := h3 1 x,
+  rw [one_inv, one_mul] at h4,
+  exact h4,
+end
```

Comment:
You can get to this result a lot faster with:
```suggestion
begin
  rw eq_top_iff,
  intros x _,
  have := quotient_group.eq.1 (subsingleton.elim (↑1 : quotient_group.quotient H) x),
  rwa [one_inv, one_mul] at this,
end
```

Reply:
Or maximally golfed:
```suggestion
top_unique $ λ x _,
  have this : 1⁻¹ * x ∈ H := quotient_group.eq.1 (subsingleton.elim _ _),
  by rwa [one_inv, one_mul] at this
```

Reply:
Ok thats amazing! With this do we even need ``setoid.eq_top_iff``? Also I think ``quot_by_top_is_subsingleton`` isnt being used at all, so can probably remove it unless you think there is a good reason to keep it around?